doc/doxygen/src/driver-guide.md

```diff
@@ -121,39 +121,32 @@ instead of the default params file.
 A default parameter header file for the example temperature sensor above would
 look like this (`tmpabc_params.h`):
 
-@code{.c}
-/* ... */
-
-#include "board.h"  /* THIS INCLUDE IS MANDATORY */
-#include "tmpabc.h"
-
-/* ... */
-
-/**
- * @brief   Default configuration parameters for TMPABC sensors
- * @{
- */
-#ifndef TMPABC_PARAM_I2C
-#define TMPABC_PARAM_I2C        (I2C_DEV(0))
-#endif
-#ifndef TMPABC_PARAM_ADDR
-#define TMPABC_PARAM_ADDR       (0xab)
-#endif
-
-#ifndef TMPABC_PARAMS
-#define TMPABC_PARAMS           { .i2c  = TMPABC_PARAM_I2C \
-                                  .addr = TMPABC_PARAM_ADDR }
-#endif
-/** @} */
-
-/**
- * @brief   Allocation of TMPABC configuration
- */
-static const tmpabc_params_t tmpabc_params[] = {
-    TMPABC_PARAMS
-}
-/* ... */
-@endcode
+    #include "board.h"  /* THIS INCLUDE IS MANDATORY */
```

Comment:
Why removing the `@code{.c}` directives? To me it fails to properly compile the documentation.

Reply:
It seems to work for me. @code{.c} directives seems to discard the comments while proper intendation helped.

Reply:
Can you alternatively try with `~~~~~~~~~~~~~~~~{.c}`?

Reply:
Ping @akshaim?

Reply:
@akshaim This still has issues to compile. When building the documentation I get:
```
"make" -BC doc/doxygen
make[1]: Entering directory '/home/leandro/Work/RIOT/doc/doxygen'
( cat riot.doxyfile ; echo "GENERATE_HTML = yes" ) | doxygen -
/home/leandro/Work/RIOT/doc/doxygen/src/driver-guide.md:199: warning: group drivers_tmp00x_config: ignoring title "TMP006/TMP007 Infrared Thermopile" that does not match old title "TMP006/TMP007 Infrared Thermopile Sensor driver compile configuration"

/home/leandro/Work/RIOT/doc/doxygen/src/driver-guide.md:193: warning: found subsection command outside of section context!
```

Reply:
@miri64  ~~~~~~~~~~~~~~~~ {.c} seems not to be helping either.

Reply:
> @akshaim This still has issues to compile. When building the documentation I get:
> 
> ```
> "make" -BC doc/doxygen
> make[1]: Entering directory '/home/leandro/Work/RIOT/doc/doxygen'
> ( cat riot.doxyfile ; echo "GENERATE_HTML = yes" ) | doxygen -
> /home/leandro/Work/RIOT/doc/doxygen/src/driver-guide.md:199: warning: group drivers_tmp00x_config: ignoring title "TMP006/TMP007 Infrared Thermopile" that does not match old title "TMP006/TMP007 Infrared Thermopile Sensor driver compile configuration"
> 
> /home/leandro/Work/RIOT/doc/doxygen/src/driver-guide.md:193: warning: found subsection command outside of section context!
> ```

Strange. Works for me.

Reply:
@miri64 @leandrolanzieri  Maybe I can cheat  Doxygen using code.{c} instead of code{.c}. As used in line #169

@code.{c}
/* ... */
/**
 * @brief   TMPABC sensor configuration
 * @{
 */
#define TMPABC_PARAM_I2C        (I2C_DEV(1))
#define TMPABC_PARAM_ADDR       (0x17)
/** @} */
/* ... */
@endcode

Reply:
That would be definitely wrong syntax, but if it works...

Reply:
`@code` better?

Reply:
Also: in line l92 it seems to be working?

Reply:
Ok, I tested it myself now. With `~~~~~~~~~~~~~~~ {.c}` syntax highlighting does not work, but with [this patch](https://gist.github.com/miri64/156d6fff3af99ff4d5fe803dc6b07207) I get syntax highlighting:
![image](https://user-images.githubusercontent.com/675644/86246486-5a7a7d00-bbab-11ea-884b-0e8c649f248d.png)

Reply:
@miri64  The problem is `@code{.c}` discards doxygen tags like @brief (which is expected behaviour). To show the comments I used 4 spaces indendation as stated in official [wiki](https://www.doxygen.nl/manual/markdown.html) but then the doc fails to build for @leandrolanzieri . I have now resorted to using just @code. It should work now.

Reply:
Ahhh now I understand. Then some of my review comments below are invalid, but I added fixes to them.